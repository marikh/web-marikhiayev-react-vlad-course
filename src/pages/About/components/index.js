import AboutHeroArea from './AboutHeroArea';

export {
    AboutHeroArea
}
