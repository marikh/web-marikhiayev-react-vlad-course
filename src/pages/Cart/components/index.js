import CartHeroArea from './CartHeroArea';

export {
    CartHeroArea
}
