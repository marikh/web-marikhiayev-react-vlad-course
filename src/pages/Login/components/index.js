import LoginHeroArea from './LoginHeroArea';

export {
    LoginHeroArea
}
