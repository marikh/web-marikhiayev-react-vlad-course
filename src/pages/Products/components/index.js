import ProductsHeroArea from './ProductsHeroArea';

export {
    ProductsHeroArea
}
