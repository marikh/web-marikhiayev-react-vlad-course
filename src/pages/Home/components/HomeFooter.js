import React from 'react';
import { SocialLinks } from '../../../components/';
import { withRouter } from 'react-router-dom';
import T from 'i18n-react';

export default withRouter(({ links, history, location, match }) => {
    return (
        <div>
            <SocialLinks links={links} />
            <h3 onClick={() => history.goBack()}>{T.translate("Back")}</h3>
        </div>
    )
})

