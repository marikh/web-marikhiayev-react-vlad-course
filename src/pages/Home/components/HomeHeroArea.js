import React from 'react';
import { HeroArea, Heading } from '../../../components/';
import T from 'i18n-react';

export default () => (
    <HeroArea>
        <Heading size={1}>
            {T.translate("Home")}
        </Heading>
    </HeroArea>
)
