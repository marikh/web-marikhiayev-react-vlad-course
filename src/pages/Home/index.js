import React from 'react';
import { 
  Layout, 
  Heading, 
  Section, 
  Card,
  InputField
} from '../../components/';
import { HomeFooter, HomeHeroArea }  from './components/';


export default ({ links }) => (
    <Layout 
        heroContent={HomeHeroArea}
        footerContent={HomeFooter}
    >
        <Section>
            <Heading size={2}>Services</Heading>
                <div className="" style={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'center' }}>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                </div>
            </Section>
            <Section>
                <Heading size={2}>Technologies</Heading>
                <div style={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'center' }}>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                </div>
            </Section>
            <Section>
                <Heading size={2}>Blog Posts</Heading>
                <div style={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'center' }}>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                </div>
            </Section>
            <Section>
                <Heading size={2}>Feedback</Heading>
                <div style={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'center' }}>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                  <Card>Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.</Card>
                </div>
            </Section>
    </Layout>
)