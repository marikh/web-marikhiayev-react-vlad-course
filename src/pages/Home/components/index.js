import HomeFooter from './HomeFooter';
import HomeHeroArea from './HomeHeroArea';

export {
    HomeFooter,
    HomeHeroArea    
}
