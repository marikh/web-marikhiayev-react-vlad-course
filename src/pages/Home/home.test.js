import { mount, shallow } from "enzyme";
import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import { Layout, Section } from '../../components/';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux';
import Home from '../../pages/Home/';
import {createStore} from 'redux';

const MENU_INITIAL_STATE = {showProtectedLinks : false};

const mockStore = configureStore();

const initialState = { menu : MENU_INITIAL_STATE } ;
    let store, wrapper;  

describe('Home outer tests', () => {
    

     beforeEach(()=>{
        if(wrapper != null)
            wrapper.unmount();
            
        store = mockStore(initialState);
        wrapper = mount( <Provider store={store}><MemoryRouter><Home /></MemoryRouter></Provider> );
    })

    it("has Layout", () => {
        expect(wrapper.find(Layout).length).toBeGreaterThan(0)
    })
    
    it("has Section Component", () => {
        expect(wrapper.find(Section).length).toBeGreaterThan(0)
    })

    it("layout has hero Component", () => {
        expect(wrapper.find(Layout).prop('heroContent') != null).toEqual(true)
    })

    it("layout has footer Component", () => {
        expect(wrapper.find(Layout).prop('footerContent') != null).toEqual(true)
    })
})

describe('Home Shallow Render',()=>{
    let wrapper;

    beforeEach(()=>{
        wrapper = shallow(<Home />);
    })

    it('render the DUMB component', () => {
       expect(wrapper.length).toEqual(1);
    });
})

describe('Home snapshot tests', () => {

    it('test weird snapshot', () => {

        const store = mockStore(initialState);

        const tree = renderer.create(
            <Provider store={store}>
                <MemoryRouter>
                    <Home />
                </MemoryRouter>
            </Provider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
 })

