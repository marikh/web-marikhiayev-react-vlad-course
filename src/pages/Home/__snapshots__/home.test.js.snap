// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot tests test weird snapshot 1`] = `
<div
  className="App"
  onClick={[Function]}
>
  <header
    className="App-header"
  >
    <div
      className="App-menubar"
    >
      <label
        className="hamburger-icon fa fa-bars"
        onClick={[Function]}
      />
      <div
        className="menu-content"
      >
        <div
          className="links-container"
        >
          <a
            className="page-link"
            href="/"
            onClick={[Function]}
          >
            <span>
              Home
            </span>
          </a>
          <a
            className="page-link"
            href="/about"
            onClick={[Function]}
          >
            <span>
              About
            </span>
          </a>
          <a
            className="page-link"
            href="/products"
            onClick={[Function]}
          >
            <span>
              Products
            </span>
          </a>
          <a
            className="page-link"
            href="/contact"
            onClick={[Function]}
          >
            <span>
              Contact
            </span>
          </a>
          <a
            className="page-link"
            href="/login"
            onClick={[Function]}
          >
            <span>
              SignIn
            </span>
          </a>
        </div>
        <div
          className="socialLinks"
        >
          <div
            className="social-links"
          >
            <a
              className="fa fa-facebook"
              href="facebook.com"
              target="_blank"
            />
            <a
              className="fa fa-github"
              href="github.com"
              target="_blank"
            />
            <a
              className="fa fa-envelope"
              href="vnovick@gmail.comm"
              target="_blank"
            />
          </div>
        </div>
      </div>
      <div
        className="overlay"
      />
    </div>
    <section
      className="App-subheader"
    >
      <div
        className="App-hero"
      >
        <h1
          className="heading heading-1"
        >
          Home
        </h1>
      </div>
    </section>
  </header>
  <section
    className="App-body"
  >
    <div
      className="raised-2 section"
    >
      <h2
        className="heading heading-2"
      >
        Services
      </h2>
      <div
        className=""
        style={
          Object {
            "display": "flex",
            "flexWrap": "wrap",
            "justifyContent": "center",
          }
        }
      >
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
      </div>
    </div>
    <div
      className="raised-2 section"
    >
      <h2
        className="heading heading-2"
      >
        Technologies
      </h2>
      <div
        style={
          Object {
            "display": "flex",
            "flexWrap": "wrap",
            "justifyContent": "center",
          }
        }
      >
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
      </div>
    </div>
    <div
      className="raised-2 section"
    >
      <h2
        className="heading heading-2"
      >
        Blog Posts
      </h2>
      <div
        style={
          Object {
            "display": "flex",
            "flexWrap": "wrap",
            "justifyContent": "center",
          }
        }
      >
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
      </div>
    </div>
    <div
      className="raised-2 section"
    >
      <h2
        className="heading heading-2"
      >
        Feedback
      </h2>
      <div
        style={
          Object {
            "display": "flex",
            "flexWrap": "wrap",
            "justifyContent": "center",
          }
        }
      >
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
        <div
          className="raised-2 card"
        >
          Consectetur cillum enim cillum dolor cupidatat pariatur occaecat.
        </div>
      </div>
    </div>
  </section>
  <footer
    className="App-footer"
  >
    <div>
      <div
        className="social-links"
      >
        <a
          className="text-link"
          href="facebook.com"
          target="_blank"
        >
          Facebook
        </a>
        <a
          className="text-link"
          href="github.com"
          target="_blank"
        >
          Github
        </a>
        <a
          className="text-link"
          href="vnovick@gmail.comm"
          target="_blank"
        >
          Email
        </a>
      </div>
      <h3
        onClick={[Function]}
      >
        Back
      </h3>
    </div>
  </footer>
</div>
`;
